WooCommerce/Classes/POS/Models/PointOfSaleSettingsService.swift

```diff
@@ -0,0 +1,95 @@
+import Foundation
+import struct Yosemite.SiteSetting
+import enum Yosemite.SettingAction
+import enum Yosemite.Plugin
+import class Yosemite.PluginsService
+import Observation
+
+@Observable final class PointOfSaleSettingsService {
+    private(set) var receiptStoreName: String?
+    private(set) var receiptStoreAddress: String?
+    private(set) var receiptStorePhone: String?
+    private(set) var receiptStoreEmail: String?
+    private(set) var receiptRefundReturnsPolicy: String?
+    private(set) var isLoading: Bool = false
+    private(set) var shouldShowReceiptInformation: Bool = false
+
+    private var siteID: Int64 {
+        ServiceLocator.stores.sessionManager.defaultSite?.siteID ?? 0
+    }
+
+    var storeName: String {
+        guard let site = ServiceLocator.stores.sessionManager.defaultSite else {
+            return Localization.storeNotSet
+        }
+        return site.name
+    }
+
+    var storeAddress: String {
+        SiteAddress().address
+    }
+
+    @MainActor
+    func retrievePOSReceiptSettings() async {
+        isLoading = true
+
+        shouldShowReceiptInformation = await isPluginSupported(.wooCommerce, minimumVersion: "10.0")
+
+        guard shouldShowReceiptInformation else {
+            isLoading = false
+            return
+        }
+
+        await withCheckedContinuation { (continuation: CheckedContinuation<Void, Never>) in
+            let action = SettingAction.retrievePointOfSaleSettings(siteID: siteID) { [weak self] result in
+                guard let self else { return }
+                switch result {
+                case .success(let siteSettings):
+                    updateReceiptSettings(from: siteSettings)
+                case .failure(let error):
+                    DDLogError("Failed to load POS settings: \(error)")
+                }
+                isLoading = false
+                continuation.resume()
+            }
+            ServiceLocator.stores.dispatch(action)
+        }
+    }
+
+    @MainActor
+    private func isPluginSupported(_ plugin: Plugin, minimumVersion: String) async -> Bool {
+        let storageManager = ServiceLocator.storageManager
```

Comment:
nit: let's DI the storage manager with `ServiceLocator.storageManager` as the default value for testability as well.

Reply:
Done! https://github.com/woocommerce/woocommerce-ios/pull/16034/commits/b102193dab0307673a244d15db9a6009ee2a44d9 I'm wondering if we should inject the whole PluginService, but I left this one out for now.